### PR TITLE
Reduce overhead in health_checker_impl by fixing alignment

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message:  Reduce overhead in health_checker_impl by fixing alignment
Additional Description: Moves many of the smaller members of the class to the end to save small amounts of memory
Risk Level: Low

### CODE REVIEW

File: source/common/upstream/health_checker_impl.cc [hunk 1]
```diff
@@ -302,10 +302,12 @@ HttpHealthCheckerImpl::HttpActiveHealthCheckSession::HttpActiveHealthCheckSessio
     : ActiveHealthCheckSession(parent, host), parent_(parent),
       response_body_(std::make_unique<Buffer::OwnedImpl>()),
       hostname_(getHostname(host, parent_.host_value_, parent_.cluster_.info())),
-      protocol_(codecClientTypeToProtocol(parent_.codec_client_type_)),
+
       local_connection_info_provider_(std::make_shared<Network::ConnectionInfoSetterImpl>(
           Network::Utility::getCanonicalIpv4LoopbackAddress(),
-          Network::Utility::getCanonicalIpv4LoopbackAddress())) {}
+          Network::Utility::getCanonicalIpv4LoopbackAddress())),
+      protocol_(codecClientTypeToProtocol(parent_.codec_client_type_)), expect_reset_{},
```
Comment (adisuissa) [inline source/common/upstream/health_checker_impl.cc:309]:
```suggestion
      protocol_(codecClientTypeToProtocol(parent_.codec_client_type_)), expect_reset_(false),
```

File: source/common/upstream/health_checker_impl.h [hunk 2]
```diff
@@ -206,11 +206,11 @@ class HttpHealthCheckerImpl : public HealthCheckerImplBase {
     Http::ResponseHeaderMapPtr response_headers_;
     Buffer::InstancePtr response_body_;
     const std::string& hostname_;
-    const Http::Protocol protocol_;
     Network::ConnectionInfoProviderSharedPtr local_connection_info_provider_;
-    bool expect_reset_{};
-    bool reuse_connection_ = false;
-    bool request_in_flight_ = false;
+    const Http::Protocol protocol_;
```
Comment (adisuissa) [inline source/common/upstream/health_checker_impl.h:210]: nit: please add a small note about the coalescing of these data members at the end (something like [this](https://github.com/envoyproxy/envoy/blob/1ac0a5948ca317fd418777dab2ffe27935c0a276/source/common/router/config_impl.h#L342)).

**Review by adisuissa — COMMENTED** [review]
Thanks! Left a couple of nits.

File: source/common/upstream/health_checker_impl.h (hunk 2 above)
Comment (adisuissa) [inline source/common/upstream/health_checker_impl.h:210]: Can the type of [Protocol](https://github.com/envoyproxy/envoy/blob/1ac0a5948ca317fd418777dab2ffe27935c0a276/envoy/http/protocol.h#L12) be redefined?

File: source/common/upstream/health_checker_impl.cc (hunk 1 above)
Comment (DiazAlan) [reply source/common/upstream/health_checker_impl.cc]: Done

File: source/common/upstream/health_checker_impl.h (hunk 2 above)
Comment (DiazAlan) [reply source/common/upstream/health_checker_impl.h:211]: Done

Comment (DiazAlan) [reply source/common/upstream/health_checker_impl.h:211]: Talked offline, about setting it to uint8. Done

**Review by zuercher — APPROVED** [review]
Thanks!

**Pull request merged**